roughpy_compute/dense/basic/shuffle_tensor_product.hpp

```diff
@@ -0,0 +1,171 @@
+#ifndef ROUGHPY_COMPUTE_DENSE_BASIC_SHUFFLE_TENSOR_PRODUCT_HPP
+#define ROUGHPY_COMPUTE_DENSE_BASIC_SHUFFLE_TENSOR_PRODUCT_HPP
+
+#include <algorithm>
+
+#include "roughpy_compute/common/basis.hpp"
+#include "roughpy_compute/common/bitmask.hpp"
+#include "roughpy_compute/common/cache_array.hpp"
+#include "roughpy_compute/common/operations.hpp"
+#include "roughpy_compute/dense/views.hpp"
+
+namespace rpy::compute::basic {
+inline namespace v1 {
```

Comment:
Would be good to summarise differences between v1 and v2 as hard to see when skimming. Tiling jumps out but not easy to see if any other differences.